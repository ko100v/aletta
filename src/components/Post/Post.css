.title {
  font-size: 3rem;
}

.content {
  font-size: 1.8rem;
  line-height: 2;
  word-wrap: break-word;
}

.link {
  color: #2F89DE;
}

.link:focus,
.link:hover {
  color: #30699e ;
  transition: color 0.3s ease;
}