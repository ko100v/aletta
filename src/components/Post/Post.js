import React, { Component } from 'react'
import styles from './Post.css'

import { Link } from 'react-router-dom'

class Post extends Component {

  render() {
    const { title, content, link } = styles

    return (
      <div>
        <h2 className={title}>
          {this.props.titleLink
            ? <Link className={styles.link} to={`/post/${this.props.slug}`}>{this.props.title}</Link>
            : this.props.title}
        </h2>
        <p className={content}>{this.props.excerpt}</p>
      </div>
    )
  }
}

export default Post