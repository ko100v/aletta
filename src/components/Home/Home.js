import React from 'react'

import Post from '../Post/Post'

const Home = ({ posts }) => (
  <div>
    {posts.length ?
      posts.map((post) => <Post key={post.slug} {...post} titleLink />)
      : <p>Awaiting Posts!</p>}
  </div>

)

export default Home