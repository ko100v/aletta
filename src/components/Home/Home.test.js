import React from 'react'

import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import Home from './Home'
import Post from '../Post/Post'
import posts from '../../db/blog.post.json'

test('Render Home', () => {
    const component = shallow(<Home posts={posts.posts} />)
    const tree = toJson(component)
    expect(component.find(Post).length).toBe(posts.posts.length)
    expect(tree).toMatchSnapshot()
})

test('Render Home', () => {
    const component = shallow(<Home posts={[]} />)
    const tree = toJson(component)
    expect(component.find(Post).length).toBe(0)
    expect(tree).toMatchSnapshot()
})