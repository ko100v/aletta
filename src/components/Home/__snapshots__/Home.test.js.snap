// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home 1`] = `
<div>
  <Post
    content="One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on. One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on. One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on. One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on. One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on. One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on."
    excerpt="One of the easiest ways to get a job in the early GitHub days was to work on one of our open source projects. We’d start to recognize your username and ability, and you’d be able to get a better idea of how we operated internally. For a small, scrappy, bootstrapped company, this was a huge time saver for us early on."
    key="how-to-build-a-progressive-web-app-with-react"
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true}
  />
  <Post
    content="Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools. Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools. Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools. Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools. Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools. Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools."
    excerpt="Jest can be used in projects that use webpack to manage assets, styles, and compilation. webpack does offer some unique challenges over other tools because it integrates directly with your application to allow managing stylesheets, assets like images and fonts, along with the expansive ecosystem of compile-to-JavaScript languages and tools."
    key="testing-with-jest-and-enzyme"
    slug="testing-with-jest-and-enzyme"
    title="Testing with Jest and Enzyme"
    titleLink={true}
  />
  <Post
    content="A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. All URLs (url(...)) and @imports are in module request format (./xxx and ../xxx means relative, xxx and xxx/yyy means in modules folder, i. e. in node_modules). CSS Modules compile to a low-level interchange format called ICSS or Interoperable CSS, but are written like normal CSS files. A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. All URLs (url(...)) and @imports are in module request format (./xxx and ../xxx means relative, xxx and xxx/yyy means in modules folder, i. e. in node_modules). CSS Modules compile to a low-level interchange format called ICSS or Interoperable CSS, but are written like normal CSS files. A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. All URLs (url(...)) and @imports are in module request format (./xxx and ../xxx means relative, xxx and xxx/yyy means in modules folder, i. e. in node_modules). CSS Modules compile to a low-level interchange format called ICSS or Interoperable CSS, but are written like normal CSS files. A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. All URLs (url(...)) and @imports are in module request format (./xxx and ../xxx means relative, xxx and xxx/yyy means in modules folder, i. e. in node_modules). CSS Modules compile to a low-level interchange format called ICSS or Interoperable CSS, but are written like normal CSS files. A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. All URLs (url(...)) and @imports are in module request format (./xxx and ../xxx means relative, xxx and xxx/yyy means in modules folder, i. e. in node_modules). CSS Modules compile to a low-level interchange format called ICSS or Interoperable CSS, but are written like normal CSS files. "
    excerpt="A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. All URLs (url(...)) and @imports are in module request format (./xxx and ../xxx means relative, xxx and xxx/yyy means in modules folder, i. e. in node_modules). CSS Modules compile to a low-level interchange format called ICSS or Interoperable CSS, but are written like normal CSS files."
    key="getting-started-with-css-modules-in-webpack"
    slug="getting-started-with-css-modules-in-webpack"
    title="Getting started with CSS modules in Webpack"
    titleLink={true}
  />
</div>
`;

exports[`Render Home 2`] = `
<div>
  <p>
    Awaiting Posts!
  </p>
</div>
`;
